docs/001_develop/03_client-capabilities/008_interaction/file-upload.mdx

```diff
@@ -0,0 +1,328 @@
+```
+
+</td>
+    </tr>
+    <tr>
+      <td>entityId</td>
+      <td><code>string</code></td>
+      <td>This attribute specifies the unique identifier that will be used by the **uploadEventName** endpoint to upload the files. This unique identifier will also be used by **uploadedFilesResourceName** as a value to filter the files list.</td>
+<td>
+
+```typescript
+<rapid-file-upload entityId="Identifier">
+```
+
+</td>
+    </tr>
+    <tr>
+      <td>uploadKey</td>
+      <td><code>string</code></td>
+      <td>This attribute specifies a unique key that will be used by the **uploadEventName** endpoint to upload the files.</td>
+<td>
+
+```typescript
+<rapid-file-upload uploadKey="UniqueKey">
+```
+
+</td>
+    </tr>
+    <tr>
+      <td>grid-fields</td>
+      <td><code>string</code></td>
+      <td>This attribute specifies the name of fields that you want to display as grid columns from the metadata of the `data-grid` datasource. By default all the fields from `data-grid` datasource metadata will be displayed. They must be specified in the format: *FILE_NAME DOWNLOAD LAST_UPDATED_BY LAST_UPDATED_AT*. The ordering which the fields are specified, will be reflected in the `data-grid` columns. If you want to display the **Download** button in the grid, you must include the **DOWNLOAD** field in the **grid-fields** attribute list along with **downloadEventName** attribute.</td>
+<td>
+
+```typescript
+<rapid-file-upload grid-fields="FILE_NAME DOWNLOAD LAST_UPDATED_BY">
+```
+
+</td>
+    </tr>
+    <tr>
+      <td>downloadEventName</td>
+      <td><code>string</code></td>
+      <td>This attribute is an `observable` of type `function`. When you click the `download` button from a row in `data-grid`, it sends the `rowData` object as a parameter to the **downloadEventName** function. It should return the name of the endpoint that will be used to download the file. This is required in order to enable the download functionality.</td>
+<td>
+
+```typescript
+<rapid-file-upload downloadEventName="DOWNLOAD_EVENT">
+```
+
+</td>
+    </tr>
+  </tbody>
+</table>
+
+### Properties
+
+<table>
+  <thead><tr><th>Name</th><th>Type</th><th>Description</th><th>Example</th></tr></thead>
+<tbody>
+<tr>
+  <td>filesGridColumnDefinitions</td>
+  <td><code>string</code></td>
+  <td>This property is used to override the colum definitions used in the attached grid</td>
+<td>
+
+```typescript
+<rapid-file-upload label="Upload file">
+```
+
+</td>
+</tr>
+</tbody>
+</table>
+
+### Slots
+| Slot | Use |
+|---|---|
+| end | slot for clear icon |
+
+### Parts
+| Part | Use |
+|---|---|
+| file-name-input | The text field showing the selected file. |
+| select-file-button | The select file button. |
+| upload-file-button | The file upload button. |
+| uploaded-files-grid | The uploaded files grid. |
+
+### Events fired
+This component doesnt fire any events.
+
+### Events listened to
+| Event | Use |
+|---|---|
+| errorOut | Emits when there is an error in the file upload. |
```

Comment:
It's emitting this event, so it goes in the fired section. Also write out the full table using the example like 
```
<rapid-file-upload @errorOut=${(x, ctx) => x.handleError(ctx.event)}
```